codec/src/main/java/io/netty/handler/codec/DelimiterBasedFrameDecoder.java

```diff
@@ -311,8 +311,13 @@ private void fail(long frameLength) {
      * Returns the number of bytes between the readerIndex of the haystack and
      * the first needle found in the haystack.  -1 is returned if no needle is
      * found in the haystack.
+     *
+     * @param haystack the buffer to search for an occurence of the needle
+     * @param needle the buffer to find in the haystack
+     *
+     * @return the first relative byte index of the needle found within the haystack, or -1 if not found
      */
-    private static int indexOf(ByteBuf haystack, ByteBuf needle) {
+    protected static int indexWithin(ByteBuf haystack, ByteBuf needle) {
```

Comment:
I'm not entirely sure, but the behavior is definitely different.  For my PR, I did nothing but rename this method since I found the previous name confusing (but would be happy to change it back).  If I try switching out the invocation of this method with `ByteBufUtil.indexOf`, (and flip the args, since they are in the opposite order for that method), the tests no longer pass and I haven't yet had the time to dig into why.